iceoryx_posh/include/iceoryx_posh/popo/base_subscriber.hpp

```diff
@@ -83,31 +83,49 @@ class BaseSubscriber
     ///
     void unsubscribe() noexcept;
 
+    // iox-#408 replace
     ///
     /// @brief hasData Check if sample is available.
     /// @return True if a new sample is available.
     ///
     bool hasSamples() const noexcept;
 
+    // iox-#408 replace
     ///
     /// @brief hasMissedSamples Check if samples have been missed since the last hasMissedSamples() call.
     /// @return True if samples have been missed.
     /// @details Samples may be missed due to overflowing receive queue.
     ///
     bool hasMissedSamples() noexcept;
 
+    // iox-#408 remove
     ///
     /// @brief take Take the a sample from the top of the receive queue.
     /// @return An expected containing populated optional if there is a sample available, otherwise empty.
     /// @details The memory loan for the sample is automatically released when it goes out of scope.
     ///
-    cxx::expected<cxx::optional<Sample<const T>>, ChunkReceiveError> take() noexcept;
+    cxx::expected<cxx::optional<Sample<const T>>, ChunkReceiveResult> take() noexcept;
 
+    ///
+    /// @brief takeChunk Take the chunk from the top of the receive queue.
+    /// @return The header of the chunk taken.
+    /// @details No automatic cleaunp of the associated chunk is performed.
+    ///
+    cxx::expected<const mepoo::ChunkHeader*, ChunkReceiveResult> takeChunk() noexcept;
```

Comment:
why do we have a `takeChunk` with `ChunkHeader*` and a `relaeseChunk` with `void*`. Should be symmetric here, or?

Reply:
The idea was that this is just the base and not exposed to the user. But releaseChunk in base at least should be symmetric, right. I will reorganize it somewhat.

Reply:
It would be the best to convert only once and as late as possible(little efficiency gain).. It makes no sense to convert back and forth between header and payload and the low level API provides the header.